props/prop_base.h

```diff
@@ -958,6 +958,25 @@ class PropBase : CommandParser, Looper, protected SaberBase {
         }
         break;
       }
+      case SaberBase::COLOR_CHANGE_MODE_ZOOMED: {
+#define ZOOM_ANGLE (M_PI * 2 / 2000)
+        float a = fusor.angle2() - current_tick_angle_;
+        if (a > M_PI) a-=M_PI*2;
+        if (a < -M_PI) a+=M_PI*2;
+        if (a > ZOOM_ANGLE * 2/3) {
+          current_tick_angle_ += ZOOM_ANGLE;
+          if (current_tick_angle_ > M_PI) current_tick_angle_ -= M_PI * 2;
+          STDOUT << "TICK+\n";
+          SaberBase::UpdateVariation(1);
+        }
+        if (a < -ZOOM_ANGLE * 2/3) {
+          current_tick_angle_ -= ZOOM_ANGLE;
+          if (current_tick_angle_ < M_PI) current_tick_angle_ += M_PI * 2;
+          STDOUT << "TICK-\n";
+          SaberBase::UpdateVariation(-1);
```

Comment:
This can make the variation negative.
For "ticked" variations that's not an issue, but I'm not sure how well it works for smooth variations.
(The  code for COLOR_CHANGE_SMOOTH keeps the values in the 0 - 32768 range.)

Reply:
It was based on STEPPED, should we apply a clamp in SaberBase::UpdateVariation to prevent negative value?